GoogleForCreators/web-stories-wp
# Keep styling when clearing rich text

## Summary

* When a rich text field is rendered or modified (and still non-empty), store current inline formatting in a ref.
* When a rich text field is modified and empty, restore inline text formatting from stored value in ref.

## Relevant Technical Choices

_None_.

## To-do

* [x] Store and restore inline text formatting when clearing a rich text
* [x] Add karma tests including regression

## User-facing changes

* When clearing a text field, the text styling persists.

## Testing Instructions

1. Add a text field
1. Make it bold
1. Click the text field to edit
1. Press backspace to remove contents
1. Start typing and observe that the text is still bold.

---



Fixes #1774.
